### PR TITLE
ig: add package v0.25.0

### PR DESCRIPTION
**Pull request by eiffel-fl**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR adds ig, the CLI flavor of [Inspektor Gadget](https://github.com/inspektor-gadget/inspektor-gadget) to Mariner.

###### Change Log  

- Add ig v0.25.0

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- TODO

###### Information

Same as #7777 but for `2.0`.

### CODE REVIEW

**Review by mfrw — COMMENTED** [review]
Can we please retarget this to `main`

File: SPECS/ig/ig.spec [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+%description
+Inspektor Gadget is a collection of tools (or gadgets) to debug and inspect Kubernetes resources and applications. It manages the packaging, deployment and execution of eBPF programs in a Kubernetes cluster, including many based on BCC tools, as well as some developed specifically for use in Inspektor Gadget. It automatically maps low-level kernel primitives to high-level Kubernetes resources, making it easier and quicker to find the relevant information.
+
+This package contains ig, the local CLI flavor of Inspektor Gadget.
+
+%prep
+%autosetup -n inspektor-gadget-%{version}
+%setup -q -n inspektor-gadget-%{version} -T -D -a 1
```
Comment (PawelWMS) [inline SPECS/ig/ig.spec:22]: Is there a reason we need both `%autosetup` and `%setup`? Usually, `%autosetup` is sufficient.

File: SPECS/ig/ig.spec [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+make gadgets-unit-tests
+ig_file=$(mktemp /tmp/ig-XXXXXX.out)
+sudo ./bin/build/ig trace exec --host > $ig_file &
+ig_pid=$!
+sleep inf &
+sleep_pid=$!
+kill $ig_pid
+kill $sleep_pid
+grep -P "${sleep_pid}\s+\d+\s+sleep" $ig_file
+rm $ig_file
```
Comment (PawelWMS) [inline SPECS/ig/ig.spec:46]: What are the instruction after `make gadgets-unit-tests` doing? Could you add a comment with some explanation? Also, when is this considered a passing test, so we fail or pass correctly?

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 3]
```diff
@@ -2187,6 +2187,7 @@
                 "helm",
                 "hvloader",
                 "hvloader-signed",
+                "ig",
```
Comment (PawelWMS) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:2190]: Why do we want to add this package? I've heard the name "Inspektor Gadget" floating around but I can't find the right deliverable or task right now.

Comment (mfrw) [reply SPECS/LICENSES-AND-NOTICES/data/licenses.json:2190]: We do have a workitem for it: [link](https://dev.azure.com/microsoft/OS/_workitems/edit/49010795)

File: SPECS/ig/ig.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/ig/ig.spec:22]: The above is another way to handle vendored dependencies.
The first autosetup untars Source0, while line 22 takes care of Source1

**Review by mfrw — APPROVED** [review]
LGTM

File: SPECS/ig/ig.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS/ig/ig.spec]: Regarding `set -e` - I wish that was the case, but at the moment it's not. The test fails if the last command returns with an exit code != 0. I'm fine with the additional testing but please also keep the missing `set -e` in mind.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
We'd need to update the `%check` section a bit to properly fail when any of the checks fail. See me other comment around the `%check` section for details.

Also, we'd need to apply any changes to `3.0-dev`, since I see #7777 has already been merged.

File: SPECS/ig/ig.spec [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+%install
+mkdir -p "%{buildroot}/%{_bindir}"
+install -D -m0755 bin/build/ig %{buildroot}/%{_bindir}
+
+%check
+set -e
+sleep inf &
+sleep_pid=$!
+./bin/build/ig snapshot process --host > $ig_file | grep -qP "sleep\s+${sleep_pid}"
+kill $sleep_pid
```
Comment (PawelWMS) [inline SPECS/ig/ig.spec:41]: What happened to `make gadgets-unit-tests`?:)

File: SPECS/ig/ig.spec [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+%install
+mkdir -p "%{buildroot}/%{_bindir}"
+install -D -m0755 bin/build/ig %{buildroot}/%{_bindir}
+
+%check
+set -e
+sleep inf &
+sleep_pid=$!
+./bin/build/ig snapshot process --host > $ig_file | grep -qP "sleep\s+${sleep_pid}"
```
Comment (PawelWMS) [inline SPECS/ig/ig.spec:40]: Where is `ig_file` set? And if we're redirecting to it, what is `grep` reading?

File: SPECS/ig/ig.spec (hunk 4 above)
Comment (PawelWMS) [reply SPECS/ig/ig.spec]: Thanks for the thorough explanation. Could you only add a comment to indicate that we're aware of the unit tests but can't run them for the reasons you've mentioned?

File: SPECS/ig/ig.spec [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+			-extldflags '-static'" \
+		-tags "netgo" \
+		-o ./bin/build/ig ./cmd/ig
+
+%install
+mkdir -p "%{buildroot}/%{_bindir}"
+install -D -m0755 bin/build/ig %{buildroot}/%{_bindir}
+
+%check
+set -e
```
Comment (PawelWMS) [inline SPECS/ig/ig.spec:37]: To be sure we fail when one of the commands fail:
```suggestion
set -e
set -o pipefail
```

**Review by PawelWMS — APPROVED** [review]
Please also port the updates to `3.0-dev`. You'll also need to replace all references to "Mariner" with "Azure Linux" in that port!

File: SPECS/ig/ig.spec [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+kill $sleep_pid
+
+%files
+%license LICENSE
+%license LICENSE-bpf.txt
+%{_bindir}/ig
+
+%changelog
+* Tue Mar 14 2023 Francis Laniel <flaniel@linux.microsoft.com> - 0.25.0-1
+- Original version for Azure Linux
```
Comment (PawelWMS) [inline SPECS/ig/ig.spec:64]: Sorry for the trouble. For now this still has to say `CBL-Mariner` - we'll need to backport the 3.0 update, which allows `Azure Linux`. If you don't want to wait for the PR check backport, then for now it's OK to use the old name.

**Comment by eric-desrochers** [comment]
@eiffel-fl The Mariner Stable Release concluded that this should be targeted against Azure Linux 3.0 (next major release), but we won't be able to accept this change for our current stable release (Mariner 2.0).

**Pull request closed without merging**